libvips/foreign/svgload.c

```diff
@@ -390,15 +456,10 @@ vips_foreign_load_svg_is_a_buffer( const void *buf, size_t len )
 		return( 0 );
 	for( i = 0; i < 24; i++ )
 		if( !isascii( str[i] ) )
-		return( 0 );
+			return( 0 );
 
 	for( i = 0; i < 200 && i < len - 5; i++ ) {
-		char txt[5];
-
-		/* 5, since we include the \0 at the end.
-		 */
-		vips_strncpy( txt, buf + i, 5 );
-		if( strcasecmp( txt, "<svg" ) == 0 )
+		if( strncmp( (const char *) str + i, "<svg", 4 ) == 0 )
```

Comment:
@jcupitt Is this change ok?

I removed the extra copy, because I only replace `str` in the SVGZ block, but not `buf`.

I switched to `strncmp` so `<SVG` also gets detected, because tag names are case insensitive.

Is casting a pointer in a loop a bad idea or will the compiler be smart about that?